include/boost/geometry/algorithms/detail/overlay/get_clusters.hpp

```diff
@@ -0,0 +1,302 @@
+    {
+        // Inner loop, walks until too far away in y-direction
+        for (auto it2 = it1 + 1; it2 != points.end(); ++it2)
+        {
+            auto const d = geometry::get<1>(it1->pnt) - geometry::get<1>(it2->pnt);
+            if (equal_policy.exceeds(d))
+            {
+                // Too far away in y-direction to be considered as equal
+                break;
+            }
+            cnt++;
+            if (equal_policy.equals(it1->pnt, it2->pnt))
+            {
+                std::size_t cindex = 0;
+                // Most recent clusters (with this y-value) are at the bottom
+                // therefore we can stop as soon as the y-value is out of reach (TODO)
+                bool found = false;
+                for (auto cit = clustered_points.begin(); cit != clustered_points.end(); ++cit)
+                {
+                    found = equal_policy.equals(cit->pnt, it1->pnt);
+                    if (found)
+                    {
+                        break;
+                    }
+                    cindex++;
+                }
+
+                // Add new cluster
+                if (! found)
+                {
+                   cindex = clustered_points.size();
+                   cluster_type newcluster;
+                   clustered_points.push_back({newcluster, it1->pnt});
+                }
+                clustered_points[cindex].cluster.turn_indices.insert(it1->turn_index);
+                clustered_points[cindex].cluster.turn_indices.insert(it2->turn_index);
+            }
+        }
+    }
+
+    // Convert to map
+    signed_size_type cluster_id = 1;
+    for (auto& trace : clustered_points)
+    {
+        clusters[cluster_id++] = trace.cluster;
+    }
+}
+
+template
+<
+    typename Turns,
+    typename Clusters
+>
+inline void assign_cluster_ids(Turns& turns, Clusters const& clusters)
+{
+    for (auto& turn : turns)
+    {
+        turn.cluster_id = -1;
+    }
+    for (auto const& kv : clusters)
+    {
+        for (const auto& index : kv.second.turn_indices)
+        {
+            turns[index].cluster_id = kv.first;
+        }
+    }
+}
+
+template <typename Geometry0, typename Geometry1>
+inline bool is_consecutive(segment_identifier const& first, segment_identifier const& second,
+                           Geometry0 const& geometry0, Geometry1 const& geometry1)
+{
+    if (first.source_index == second.source_index
+        && first.ring_index == second.ring_index
+        && first.piece_index == second.piece_index
+        && first.multi_index == second.multi_index)
+    {
+        auto const sd = first.source_index == 0
+                        ? segment_distance(geometry0, first, second)
+                        : segment_distance(geometry1, first, second);
+        return sd == 1;
+    }
+    return false;
+}
+
+
+template <typename Turn, typename Geometry0, typename Geometry1>
+inline bool corresponding_turn(Turn const& turn, Turn const& start_turn,
```

Comment:
Could this function be renamed? It's not clear in what way the turns correspond to each other.

Reply:
I think the name is OK but I understand your question.
Note that the turns with `method_start` are additional. They were not necessary in rescaling, but they are added because sometimes (rare cases) the other turn (`method_touch` or `method_touch_interior`) is missing. Therefore I call it the corresponding turn.

Reply:
:heavy_check_mark: added description to the function